internal/integratedservices/services/securityscan/spec.go

```diff
@@ -52,6 +53,10 @@ func (s integratedServiceSpec) Validate(pipelineNamespace string) error {
 		validationErrors = errors.Combine(validationErrors, s.Registry.Validate())
 	}
 
+	for _, registryItem := range s.Registries {
```

Comment:
Since we are deprecating `registry`, the above if should include a condition for `registries` being empty.

Reply:
https://github.com/banzaicloud/pipeline/pull/3403/commits/d7aa305ca101d839dc983d9160cc7a6eef964758